src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2093,23 +2002,27 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             Debug.Assert(!IsConditionalState);
 
             VisitLvalue(node.Left);
-            Result left = this.State.Result;
+            Result left = _result;
 
             VisitRvalue(node.Right);
-            Result right = this.State.Result;
+            Result right = _result;
 
             // byref assignment is also a potential write
             if (node.RefKind != RefKind.None)
             {
                 WriteArgument(node.Right, node.RefKind, method: null);
             }
 
-            TrackNullableStateForAssignment(node.Left, left.Slot, left.Type, node.Right, right.Type, right.Slot);
-
-            //if (this.State.Reachable) // PROTOTYPE(NullableReferenceTypes): Consider reachability?
+            if (node.Left.Kind == BoundKind.EventAccess && ((BoundEventAccess)node.Left).EventSymbol.IsWindowsRuntimeEvent)
```

Comment:
>IsWindowsRuntimeEvent [](start = 101, length = 21)

What about regular not field-like events?

Reply:
Assignment of non-field-like events uses `BoundEventAssignmentOperator`.

Reply:
> Assignment of non-field-like events uses BoundEventAssignmentOperator.

Consider adding a comment.

---
In reply to: [169229495](https://github.com/dotnet/roslyn/pull/24917#discussion_r169229495) [](ancestors = 169229495)